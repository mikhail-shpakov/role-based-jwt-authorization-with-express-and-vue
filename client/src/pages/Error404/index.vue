<template lang="pug" functional>
.hero.is-medium
  .hero-body
    .columns
      .column.img-404
      .column.is-one-third
        p.title-404.has-text-dark 404
        p.subtitle-404.has-text-dark Страница не найдена
        .tip-404 Возможно, она была удалена, или ее адрес изменился
        b-button.is-success(
          tag="router-link"
          :to="{ path: '/' }"
        ) Вернуться на главную
</template>

<style lang="sass" scoped>
div
  &.columns
    margin: 0 auto
    max-width: $display-width

    .title-404
      font-size: 4em
      @media (min-width: $display-bp-mobile)
        font-size: 6em

    .subtitle-404
      font-size: 1.3em
      @media (min-width: $display-bp-mobile)
        font-size: 1.5em
        margin: -5px 0 5px

    .tip-404
      margin: 0 auto 15px
      color: $color-alt

    .column
      margin: auto
      &.img-404
        background-image: url("../../assets/404.svg")
        background-size: 100% 100%
        min-height: 350px
        @media (min-width: $display-bp-mobile)
          order: 1
</style>
