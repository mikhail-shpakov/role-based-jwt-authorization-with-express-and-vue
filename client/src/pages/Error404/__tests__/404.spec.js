import { createLocalVue, RouterLinkStub, shallowMount } from '@vue/test-utils'
import Buefy from 'buefy'
import Error404 from '../index'

describe('Error404', () => {
  const localVue = createLocalVue()
  localVue.use(Buefy)
  localVue.component('router-link', RouterLinkStub)

  it('when load page, renders correctly', () => {
    const wrapper = shallowMount(Error404, {
      localVue
    })

    expect(wrapper).toMatchSnapshot()
  })
})
