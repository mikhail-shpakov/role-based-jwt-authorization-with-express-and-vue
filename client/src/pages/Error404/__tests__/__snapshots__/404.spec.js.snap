// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 when load page, renders correctly 1`] = `
<div class="hero is-medium">
  <div class="hero-body">
    <div class="columns">
      <div class="column img-404"></div>
      <div class="column is-one-third">
        <p class="title-404 has-text-dark">404</p>
        <p class="subtitle-404 has-text-dark">Страница не найдена</p>
        <div class="tip-404">Возможно, она была удалена, или ее адрес изменился</div><a type="button" class="button is-success">
          <!----> <span>Вернуться на главную</span>
          <!----></a>
      </div>
    </div>
  </div>
</div>
`;
